activestorage/lib/active_storage/attached/macros.rb

```diff
@@ -34,19 +34,36 @@ def #{name}
         end
 
         def #{name}=(attachable)
-          #{name}.attach(attachable)
+          @active_storage_attached_#{name}_change =
+            if attachable.nil?
+              ActiveStorage::Attached::Changes::DeleteOne.new("#{name}", self)
+            else
+              ActiveStorage::Attached::Changes::CreateOne.new("#{name}", self, attachable)
+            end
+        end
+
+        def #{name}_change
+          @active_storage_attached_#{name}_change
+        end
+
+        def clear_#{name}_change
+          @active_storage_attached_#{name}_change = nil
         end
       CODE
 
-      has_one :"#{name}_attachment", -> { where(name: name) }, class_name: "ActiveStorage::Attachment", as: :record, inverse_of: :record, dependent: false
+      has_one :"#{name}_attachment", -> { where(name: name) }, class_name: "ActiveStorage::Attachment", as: :record, inverse_of: :record, dependent: :destroy
```

Comment:
Does this need a deprecation cycle?

Reply:
No, it doesn’t change behavior. We did this before by calling `purge_later` or `detach` on destroy.

Reply:
Got it, so it's just more obvious now.